### PR TITLE
[UI] Registry UI issues

### PR DESCRIPTION
**Pull request by Yashsharma1911**
**Notes for Reviewers**

This PR fixes #

- [X] Fix expand and collapse all icons
- [X] Fill search automatically when using deep link
- [X] Use custom tooltip
- [x] Change tree background
- [x] Show only what is selected in the details section
- [X] Show sub-category for components
- [X] Remove redundant code and improve reusability
- [x] Make registrant as a chip
- [X] Group relationships by kind and show in tree form 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

File: ui/assets/icons/CollapseAll.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+const CollapseAllIcon = ({ width = 24, height = 24, fill = "currentColor", style = {} }) => (
```
Comment (sudhanshutech) [inline ui/assets/icons/CollapseAll.js:3]: will you also parallelly put these icons in sistent too

Comment (leecalcote) [reply ui/assets/icons/CollapseAll.js:3]: @sudhanshutech that is such a great request and such a terrible one both at the same time. Why?

File: ui/components/MeshModelRegistry/MesheryTreeView.js [hunk 2]
```diff
@@ -465,22 +469,28 @@ const MesheryTreeView = ({
...
+                disableRipple
+                disabled={disabledExpand()}
+              >
+                <ExpandAllIcon height={15} width={15} />
               </IconButton>
-            </Tooltip>
+            </CustomTextTooltip>
 
-            <Tooltip title="Collapse All" placement="top">
+            <CustomTextTooltip title="Collapse All" placement="top" backgroundColor="#3C494F">
```
Comment (leecalcote) [inline ui/components/MeshModelRegistry/MesheryTreeView.js:483]: Use the theme.

File: ui/components/MeshModelRegistry/MesheryTreeView.js [hunk 3]
```diff
@@ -465,22 +469,28 @@ const MesheryTreeView = ({
       <div>
         {width < 1370 && isSearchExpanded ? null : (
           <div style={{ display: 'flex', flexDirection: 'row' }}>
-            <Tooltip title="Expand All" placement="top">
-              <IconButton onClick={expandAll} size="large" disableRipple>
-                <KeyboardArrowDownIcon />
+            <CustomTextTooltip title="Expand All" placement="top" backgroundColor="#3C494F">
```
Comment (leecalcote) [inline ui/components/MeshModelRegistry/MesheryTreeView.js:472]: theme reference needed.

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: ui/components/MeshModelRegistry/MesheryTreeView.js (hunk 2 above)
Comment (Yashsharma1911) [reply ui/components/MeshModelRegistry/MesheryTreeView.js:483]: Thankyou for pointing out, I was testing color and forget to update with theme before pushing

**Comment by Yashsharma1911** [comment]
Moving forward here, with a note to improve things in the future

- [X] Showcasing overlapping relationships
- [X] Add adapters in registrant tab

Update relationship tab:

Screenshot 2024-02-06 at 2 50 34 AM

Issue with this tab: 

Screenshot 2024-02-06 at 2 51 24 AM

**Pull request merged**